fix insufficient precision issue in T+dT syntax

tnakazato opened the pull request:
I noticed that timerange selection by "T+dT" syntax omits the sub-seconds value. More specifically, if we write "HH:MM:SS.ss", ".ss" part is omitted in parsing "T+dT" string. I think this is a bug because "T1~T2" syntax handles sub-seconds value properly. This pull request is possible fix for the bug.

tammojan commented:
Well spotted, thanks for the fix! Could you perhaps add a test for this? This could be done by altering `tMSTimeGram.run` (copying the structure from `tMSSelection.run`), and making a reference output `tMSTimeGram.out`. The measurement set `mssel_test_small.ms` should contain enough time slots to come up with a useful test, in particular it has 23 time slots spaced 1.92 seconds apart.

tnakazato commented:
Hi, @tammojan. I will try adding the test. By the way, I pushed further improvement using MVEpoch.

tnakazato commented:
Added minimal set of verification tests in tMSTimeGram. The test checks if T+dT" works properly. As a reference, "T1~T2" test, which selects effectively the same time range, is also added. I confirmed that the test passes on the branch while fails with the code before fix.

tnakazato commented:
Any update on this? Please let me know if you have more suggestion. Thanks!

tammojan commented:
Thank you, this change looks good to me. Also good to have TimeGram tested now!

tnakazato commented:
Thanks!
